parser/impl/parser.cpp

```diff
@@ -469,17 +469,201 @@ sPtr<ast::IfStatement> Parser::parseIfStatement() {
 // ---- @aabounegm
 
 sPtr<ast::Expression> Parser::parseExpression() {
-    // TODO: Concrete implementation
+    return parseBinaryExpression(0);
```

Comment:
Should a generic `parseExpression` check first if it is only a `UnaryExpression`? There might not be another operand

Reply:
But `BinaryExpression` checks firstly for unary part, thus all operands are readen

Reply:
Yes, but `parseBinaryExpression` returns a `BinaryExpression` object. It wouldn't make much sense to consider `5` a binary expression

Reply:
1. `parseBinaryExpression` can return `UnaryExpression`
2. Yes, it is not the best idea
3. But. Why should check for unary expression if it will be check by binary one
4. It was made this way in the implementation of `GO`